Sources/Confidence/EventSenderEngine.swift

```diff
@@ -0,0 +1,97 @@
+import Combine
+import Foundation
+
+protocol EventsUploader {
+    func upload(request: EventBatchRequest) -> Bool
```

Comment:
is this actually a blocking task or should we declare it `async`?

Reply:
no, it shouldn't be async as users can be expected to call this any time and not being blocked.

Reply:
What I mean is that I would expect consumers of this api to do `let result = await upload(...)`, no?

Reply:
ah yes. i was confused by event sender engine api send function. you are right.